Add docs for `snforge new` command (#2772)
foundry-rs/starknet-foundry: docs/src/appendix/snforge/init.md

## `-h`, `--help`

Print help.

> ⚠️ **Warning**
>

> > The `snforge init` command has been deprecated. Please use the [`snforge new`](./new.md) command instead.

Reviewer: ```suggestion
> The `snforge init` command is deprecated. Please use the [`snforge new`](./new.md) command instead.
```
